### PR TITLE
fix: Add namespace for each builtin plugin

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. use kebab case for naming plugin, just follow https://github.com/rolldown/vite/blob/3bf86e3f715c952a032b476b60c8c869e9c50f3f/packages/vite/src/node/plugins/dynamicImportVars.ts#L174-L174

### CODE REVIEW

File: crates/rolldown_plugin_transform/src/lib.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub struct TransformPlugin {
 /// only handle ecma like syntax, `jsx`,`tsx`,`ts`
 impl Plugin for TransformPlugin {
   fn name(&self) -> Cow<'static, str> {
-    Cow::Borrowed("module_preload_polyfill")
+    Cow::Borrowed("transform")
```
Comment (hyf0) [inline crates/rolldown_plugin_transform/src/lib.rs]: We should have a convention for this. `builtin:transform` would be more proper I guess.

@underfin cc

Comment (hyf0) [inline crates/rolldown_plugin_transform/src/lib.rs]:
```suggestion
    Cow::Borrowed("builtin:transform")
```

Comment (IWANABETHATGUY) [reply crates/rolldown_plugin_transform/src/lib.rs]: All plugins don't have a namespace for now, prefix a namespace is better.

**Comment by IWANABETHATGUY** [comment]
Let's prefix each built-in plugin name with `builtin`.

**Review by hyf0 — APPROVED** [review]

**Pull request merged**